spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/operations/SqsTemplate.java

```diff
@@ -244,23 +250,43 @@ private Map<String, Object> addAdditionalReceiveHeaders(SqsReceiveOptionsImpl op
 	@Override
 	protected <T> org.springframework.messaging.Message<T> preProcessMessageForSend(String endpointToUse,
 			org.springframework.messaging.Message<T> message) {
-		return FifoUtils.isFifo(endpointToUse) ? addMissingFifoSendHeaders(message) : message;
+		return FifoUtils.isFifo(endpointToUse) ? addMissingFifoSendHeaders(endpointToUse, message) : message;
 	}
 
 	@Override
 	protected <T> Collection<org.springframework.messaging.Message<T>> preProcessMessagesForSend(String endpointToUse,
 			Collection<org.springframework.messaging.Message<T>> messages) {
 		return FifoUtils.isFifo(endpointToUse)
-				? messages.stream().map(this::addMissingFifoSendHeaders).collect(Collectors.toList())
+				? messages.stream().map(message -> addMissingFifoSendHeaders(endpointToUse, message)).toList()
 				: messages;
 	}
 
-	private <T> org.springframework.messaging.Message<T> addMissingFifoSendHeaders(
+	private <T> org.springframework.messaging.Message<T> addMissingFifoSendHeaders(String endpointName,
 			org.springframework.messaging.Message<T> message) {
-		return MessageHeaderUtils.addHeadersIfAbsent(message,
-				Map.of(SqsHeaders.MessageSystemAttributes.SQS_MESSAGE_GROUP_ID_HEADER, UUID.randomUUID().toString(),
-						SqsHeaders.MessageSystemAttributes.SQS_MESSAGE_DEDUPLICATION_ID_HEADER,
-						UUID.randomUUID().toString()));
+
+		String contentBasedDedupQueueAttribute;
+		try {
+			contentBasedDedupQueueAttribute = getQueueAttributes(endpointName).get()
```

Comment:
I tend to favor `join` over `get` - it enables cleaner exception handling. But I'm ok with `get` as well.

Also, there's a policy not to use static imports please.

Reply:
Moved to join and removed static imports from the SqsTemplate.  There were a number of pre-existing static imports that I left as-is in the test.